src/main/java/sg/ncl/MainController.java

```diff
@@ -424,25 +461,41 @@ public String validateDetails(@ModelAttribute("loginForm") LoginForm loginForm,
 
     	} else {
     		// logic error not suppose to reach here
+            // possible if user fill up create new team but without the team name
+            redirectAttributes.addFlashAttribute("signupError", "There is a problem when submitting your form. Please re-enter and submit the details again.");
     		return "redirect:/signup2";
     	}
     }
 
+    /**
+     * Use when registering new accounts
+     * @param mainObject A JSONObject that contains user's credentials, personal details and team application details
+     */
     private void registerUserToDeter(JSONObject mainObject) {
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         headers.set("Authorization", AUTHORIZATION_HEADER);
 
         HttpEntity<String> request = new HttpEntity<String>(mainObject.toString(), headers);
         ResponseEntity responseEntity = restTemplate.exchange(properties.getSioRegUrl(), HttpMethod.POST, request, String.class);
+        // FIXME check if email already exists
+        // FIXME check if team name duplicate when applying
+        // FIXME check if adapter connection error
     }
 
+    /**
+     * Use when users register a new account for joining existing team
+     * @param teamName The team name to join
+     * @return
+     */
     private String getTeamIdByName(String teamName) {
+        // FIXME check if team name exists
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)